pymeasure/instruments/rigol/rigol_dg800.py

```diff
@@ -0,0 +1,167 @@
+        This means displayed voltages will be internally converted to
+        correspond that seen by an infinite load, not 50 Ohms.
+        See p. 82 of the manual for more information.""",
+        validator=truncated_discrete_set,
+        values={True: "INF", False: "50"},
+        map_values=True,
+    )
+
+    frequency = Channel.control(
+        ":SOUR{ch}:FREQ?", ":SOUR{ch}:FREQ %f", """ Control the output frequency (Hz)"""
+    )
+
+    sync = Channel.control(
+        ":OUTP{ch}:SYNC?",
+        ":OUTP{ch}:SYNC %s",
+        """Control the synchronization flag.
+          True if the channel triggers the synchronization connector and False if not """,
+        validator=truncated_discrete_set,
+        values={True: "ON", False: "OFF"},
+        map_values=True,
+    )
+
+    sine = Channel.setting(
+        ":SOUR{ch}:APPL:SIN %f,%f,%f,%f",
+        """
+        Set the waveform generator to output a sine of specified parameters
+        :param freq: (int) The frequency of the sine in Hz
+        :param ampl: (float) The peak to peak amplitude of the sine in V
+        :param offset: (float) The DC offset of the sine in V
+        :param phase: (float) The phase offset of the sine in degrees
+        """,
+    )
+    square = Channel.setting(
+        ":SOUR{ch}:APPL:SQU %f,%f,%f,%f",
+        """
+        Set the waveform generator to output a square wave of specified parameters
+        :param freq: (int) The frequency of the wave in Hz
+        :param ampl: (float) The peak to peak amplitude of the wave in V
+        :param offset: (float) The DC offset of the wave in V
+        :param phase: (float) The phase offset of the wave in degrees
+        """,
+    )
+    triangle = Channel.setting(
+        ":SOUR{ch}:APPL:RAMP %f,%f,%f,%f",
+        """
+        Set the waveform generator to output a triangle wave of specified parameters
+        :param freq: (int) The frequency of the wave in Hz
+        :param ampl: (float) The peak to peak amplitude of the wave in V
+        :param offset: (float) The DC offset of the wave in V
+        :param phase: (float) The phase offset of the wave in degrees
+        """,
+    )
+    pulse = Channel.setting(
+        ":SOUR{ch}:APPL:PULS %f,%f,%f,%f",
+        """
+        Set the waveform generator to output a pulse wave of specified parameters
+        Note that the duty cycle is specified using the duty_cycle method.
+        :param freq: (int) The frequency of the wave in Hz
+        :param ampl: (float) The peak to peak amplitude of the wave in V
+        :param offset: (float) The DC offset of the wave in V
+        :param phase: (float) The phase offset of the wave in degrees
+        """,
+    )
+
+    duty_cycle = Channel.control(
+        ":SOUR{ch}:PULS:DCYC?",
+        ":SOUR{ch}:PULS:DCYC %f",
+        """
+    Control the duty cycle of pulses out of the waveform generator.
+    :param duty_cycle: (float) The pulse's duty cycle
+    """,
```

Comment:
Here the indentation is different.

You can use formatters like ruff or black to automatically format the files according to style guides.